components/AgreementList/AgreementList.js

```diff
@@ -169,12 +155,12 @@ function AgreementList(props) {
               </p>
 
               <p>
-                <strong>{t("payment-amount")}</strong>
-                {parseFloat(ethers.utils.formatUnits(agreement.payment.amount, 18))
-                  .toFixed(2)
-                  .replace(/\.00$/, "")}{" "}
-                USD
+                  <strong>{t("payment-amount")}</strong>
+                  {agreement.payment.amount && !isNaN(agreement.payment.amount)
```

Comment:
Extract this to a function that receives the amount and returns the valid string